app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -24,11 +24,8 @@ def backend_object_name
       def refresh_entities
         href = remote_href
         if href
-          if remote_options[:url]&.start_with?('uln')
-            [api.remotes_uln_api.partial_update(href, remote_options)]
-          else
-            [api.remotes_api.partial_update(href, remote_options)]
-          end
+          # Do not consider remotes_uln_api, since the Katello server will never serve ULN!
+          [api.remotes_api.partial_update(href, remote_options)]
```

Comment:
Even though it is not really necessary we could consider using `get_remotes_api` here anyway, just to keep things more generic so this won't be overlooked in case the `get_remotes_api` use case is extended further in the future. I would then use:
```suggestion
          [api.get_remotes_api(href: href).partial_update(href, remote_options)]
```
Any opinions? I am unsure what is best.

Reply:
Since for rpm/uln it should be correct (because it is created correctly), I guess we can do that.

Reply:
I like using `remotes_api` specifically if this case should always use the Rpm remote.

Reply:
I think it makes it more obvious that there is a special case here -- that ULN remotes will never be used on smart proxies.

Reply:
Then I will leave this as is. Except for the improved comment discussed below.